service/storage_proxy.cc

```diff
@@ -6283,6 +6283,8 @@ future<bool> storage_proxy::cas(schema_ptr schema, shared_ptr<cas_request> reque
 
         auto l = co_await paxos::paxos_state::get_cas_lock(token, write_timeout);
 
+        co_await utils::get_local_injector().inject("cas_timeout_after_lock", write_timeout);
```

Comment:
I don't understand how this can reproduce the problem.
Once `get_cas_lock` finished, then `guard` is in locked state. IIUC from your explanation, the problem appears if guard is destroyed **before** `_locked = true`.

Reply:
First request will stuck here. All others will be destroyed before semaphore is locked. The test runs 10 requests in parallel while error injection is active.